reducers/views/modals.ts

```diff
@@ -17,15 +20,13 @@ function modalState(state = {}, action) {
             },
         };
     case ActionTypes.MODAL_CLOSE:
-        return {
-            ...state,
-            [action.modalId]: {
-                open: false,
-                dialogProps: action.dialogProps,
-                dialogType: action.dialogType,
-            },
-        };
+        return Object.keys(state).reduce((previousModalState: ViewsState['modals']['modalState'], currentModalId) => {
```

Comment:
Instead of looping through the state, we can use `Object.deleteProperty` to delete the key a bit easier

Reply:
you mean [Reflect.deleteProperty](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Reflect/deleteProperty) or [delete](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/delete) , But in either case we are directly mutating the object itself which i feel is slightly not a better way when we are use pure non mutating way don't you think?

Reply:
✅  Updated as per request/discussion